src/text/span.rs

```diff
@@ -155,3 +161,73 @@ impl<'a> Styled for Span<'a> {
         self
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn from_ref_str_borrowed_cow() {
+        let content = "some string";
+
+        let span = Span::from(content);
+
+        assert_eq!(span.content, Cow::Borrowed(content));
+    }
+
+    #[test]
+    fn from_string_ref_str_borrowed_cow() {
+        let content = String::from("some string");
+
+        let span = Span::from(content.as_str());
+
+        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
+    }
+
+    #[test]
+    fn from_string_owned_cow() {
+        let content = String::from("some string");
+        let content_clone = content.clone();
+
+        let span = Span::from(content);
+
+        assert_eq!(span.content, Cow::Owned::<str>(content_clone));
+    }
+
+    #[test]
+    fn from_ref_string_borrowed_cow() {
+        let content = String::from("some string");
+
+        let span = Span::from(&content);
+
+        assert_eq!(span.content, Cow::Borrowed(content.as_str()));
+    }
+
+    #[test]
+    fn reset_should_set_style_reset() {
+        let mut span = Span::styled("test", Style::default().fg(crate::style::Color::Green));
+
+        assert_eq!(span.style, Style::default().fg(crate::style::Color::Green));
+
+        span.reset_style();
+
+        assert_eq!(span.style, Style::reset());
+        assert_ne!(span.style, Style::default());
```

Comment:
How about simplifying this:
```suggestion
        let mut span = "test".green();
        span.reset_style();
        assert_eq!(span.style, Style::reset());
```

Reply:
i didnt know `.green()` would make it a `Span`, but i think this should be explicit, it is also just testing before the reset that it is in a known proper condition (and that a change actually happened)

Reply:
I tend to avoid precondition tests and instead test the preconditions in tests that actually care about that condition.
The concise approach seems sufficient enough. Not a blocker though.